hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/nfttokeninfo/NftTokenInfoCall.java

```diff
@@ -61,40 +65,54 @@ public NftTokenInfoCall(
         requireNonNull(token);
-        final var nft = enhancement
-                .nativeOperations()
-                .getNft(token.tokenIdOrElse(ZERO_TOKEN_ID).tokenNum(), serialNumber);
-        final var status = nft != null ? SUCCESS : ResponseCodeEnum.INVALID_TOKEN_NFT_SERIAL_NUMBER;
-        return fullResultsFor(status, gasCalculator.viewGasRequirement(), token, nft);
+        return fullResultsFor(SUCCESS, gasCalculator.viewGasRequirement(), token);
     }
 
     @Override
     protected @NonNull FullResult viewCallResultWith(
             @NonNull final ResponseCodeEnum status, final long gasRequirement) {
-        return fullResultsFor(status, gasRequirement, Token.DEFAULT, null);
+        return fullResultsFor(status, gasRequirement, Token.DEFAULT);
     }
 
     private @NonNull FullResult fullResultsFor(
-            @NonNull final ResponseCodeEnum status,
-            final long gasRequirement,
-            @NonNull final Token token,
-            @Nullable final Nft nft) {
+            @NonNull final ResponseCodeEnum status, final long gasRequirement, @NonNull final Token token) {
         requireNonNull(status);
         requireNonNull(token);
 
         final var ledgerConfig = configuration.getConfigData(LedgerConfig.class);
         final var ledgerId = Bytes.wrap(ledgerConfig.id().toByteArray()).toString();
+        final var nft = enhancement
+                .nativeOperations()
+                .getNft(token.tokenIdOrElse(ZERO_TOKEN_ID).tokenNum(), serialNumber);
         // @Future remove to revert #9074 after modularization is completed
-        if (isStaticCall && (status != SUCCESS || nft == null)) {
+        if ((isStaticCall && (status != SUCCESS)) || nft == null) {
             return revertResult(status, gasCalculator.viewGasRequirement());
         }
 
-        final var nonNullNft = nft != null ? nft : Nft.DEFAULT;
+        Account ownerAccount = getOwnerAccount(nft, token);
+        if (ownerAccount == null) {
+            return revertResult(INVALID_ACCOUNT_ID, gasCalculator.viewGasRequirement());
+        }
         return successResult(
                 NON_FUNGIBLE_TOKEN_INFO
                         .getOutputs()
                         .encodeElements(
-                                status.protoOrdinal(), nftTokenInfoTupleFor(token, nonNullNft, serialNumber, ledgerId)),
+                                status.protoOrdinal(),
+                                nftTokenInfoTupleFor(token, nft, serialNumber, ledgerId, ownerAccount)),
                 gasRequirement);
     }
+
+    private Account getOwnerAccount(Nft nft, Token token) {
+        final var explicitId = nft.ownerIdOrElse(AccountID.DEFAULT);
+        if (explicitId.account().kind() == AccountID.AccountOneOfType.UNSET) {
+            return null;
+        }
+        final long ownerNum;
+        if (explicitId.accountNumOrElse(TREASURY_OWNER_NUM) == TREASURY_OWNER_NUM) {
```

Comment:
Maybe we can directly use if(!explicitId.hasAccountNum()) and then we wouldn't need the constant TREASURY_OWNER_NUM